draft-ppm-protocol.md

```diff
@@ -934,6 +961,53 @@ https://github.com/abetterinternet/prio-documents/issues/112]]
 issue#57). For example, how does a leader respond to a collect request if the
 helper drops out?]
 
+### Validating Batch Parameters {#batch-parameter-validation}
+
+Before an aggregator responds to a collect request or output share request, it
+must first check that the request does not violate the parameters associated
+with the PPM task. It does so as described here.
+
+First the aggregator checks that the request's batch interval respects the
+boundaries defined by the PPM parameters `Param`. Namely, it checks that both
+`batch_start` and `batch_end` are divisible by `Param.min_batch_duration` and
+that `batch_end - batch_window >= min_batch_duration`. Unless both these
+conditions are true, it aborts and alerts the peer with "invalid batch
+interval".
+
+Next, the aggregator checks that the request respects the generic privacy
+parameters of the PPM task. Let `X` denote the set of input shares the
+aggregator has validated and which fall in the batch interval of the request.
+
+* If `len(X) < Param.min_batch_size`, then the aggregator aborts and alerts the
+  peer with "insufficient batch size".
+* The aggregator keeps track of the number of times each input share was added
+  to the batch of an output share request. If any input share in `X` was added
+  to at least `Param.max_report_lifetime` previous batches, then the helper
+  aborts and alerts the peer with "request exceeds the batch's privacy budget".
+
+### Anti-replay {#anti-replay}
+
+Except as required for the PPM protocol, using a report in multiple batches, or
```

Comment:
You're right, this needs to be rephrased. What do you think about

> Using a report multiple times withing a single batch, or using the same report in multiple batches, is considered a privacy violation. To prevent such attacks...